pkg/cmd/server/start/master_args.go

```diff
@@ -193,6 +189,14 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 		},
 	}
 
+	if args.BindAddrArg.BindAddr.URL.Scheme == "https" {
```

Comment:
Hmm, This is in arg so pace. We don't have flags for specifying certs, so the listen scheme is the only hook we have.